File: java/1099.java
class Solution {
    public int twoSumLessThanK(int[] A, int K) {
        int ans = -1;
        Arrays.sort(A);
        int lo = 0,hi = A.length-1;
        while(lo < hi){
            int S = A[lo] + A[hi];
            if(S < K){
                ans = Math.max(S,ans);
                lo ++;
            }else{
                hi --;
            }
        }
        return ans;
    }
}

File: go/1099.go
import "sort"

func twoSumLessThanK(A []int, K int) int {
	sort.Ints(A)
	ans := -1
	lo, hi := 0, len(A)-1
	for lo < hi {
		sum := A[lo] + A[hi]
		if sum >= K {
			hi--
		} else {
			ans = max(ans, sum)
			lo++
		}
	}
	return ans
}

func max(i, j int) int {
	if i > j {
		return i
	}
	return j
}